x/dclupgrade/keeper/msg_server_approve_upgrade.go

```diff
@@ -51,6 +51,9 @@ func (k msgServer) ApproveUpgrade(goCtx context.Context, msg *types.MsgApproveUp
 		if err != nil {
 			return nil, err
 		}
```

Comment:
`upgradeKeeper.ScheduleUpgrade` call and the check of its result must be moved to the top of the `if` branch because we must not do any changes in the store in case validation fails and the upgrade is not scheduled.